Move the CI-related scripts to dedicated directories

shindere opened the pull request:
This PR adds a CI directory containing one subdirectory per CI
solution: AppVeyor, Inria and Travis.

This is a follow-up to a discussion with @dra27

shindere committed:
Move the CI-related scripts to dedicated directories

nojb commented:
- This is nitpicking, but could we keep everything in lowercase, please?
- why not move them inside `tools/` instead of adding a new toplevel directory?

shindere commented:
Nicolás Ojeda Bär (2018/05/23 15:54 +0000):
> - This is nitpicking, but could we keep everything in lowercase, please?
> - why not move them inside `tools/` instead of adding a new toplevel
> - directory?

Followed both suggestions, thanks!
And rebased.

shindere commented:
I just rebased on latest trunk to take @xavierleroy's
4adae1b9bd3af47c1d626950433c56a3ef576747 commit into account.

I also updated the other-configs script to invoke the new main script
rather than the legacy ci-build one.

Finally, variables have been introduced to avoid repeating the call and the
-j8 option.

xavierleroy commented on tools/ci-build-other-configs.sh:
```diff
@@ -0,0 +1,5 @@
+# This is expected to be run from the toplevel of the OCaml source tree
+# This script can be removed once none of the branch tested on Inria's CI
+# expects to find it here, i.e. once we stop working on the 4.07 branch
+./tools/ci/inria/other-configs $*
```
I's rather use `"$@"` instead of `$*` in the off chance that there are spaces in the arguments.

xavierleroy commented on tools/ci-build.sh:
```diff
@@ -0,0 +1,5 @@
+# This is expected to be run from the toplevel of the OCaml source tree
+# This script can be removed once none of the branch tested on Inria's CI
+# expects to find it here, i.e. once we stop working on the 4.07 branch
+./tools/ci/inria/main $*
```
Same `"$@"` suggestion.

xavierleroy reviewed (approved):
Looks good to me, with a suggestion below.

shindere commented:
Xavier Leroy (2018/05/24 00:14 -0700):
> xavierleroy approved this pull request.
> 
> Looks good to me, with a suggestion below.
> 
> > @@ -0,0 +1,5 @@
> +#!/bin/sh
> +# This is expected to be run from the toplevel of the OCaml source tree
> +# This script can be removed once none of the branch tested on Inria's CI
> +# expects to find it here, i.e. once we stop working on the 4.07 branch
> +./tools/ci/inria/other-configs $*
> 
> I's rather use `"$@"` instead of `$*` in the off chance that there are spaces in the arguments.
> 
> > @@ -0,0 +1,5 @@
> +#!/bin/sh
> +# This is expected to be run from the toplevel of the OCaml source tree
> +# This script can be removed once none of the branch tested on Inria's CI
> +# expects to find it here, i.e. once we stop working on the 4.07 branch
> +./tools/ci/inria/main $*
> 
> Same `"$@"` suggestion.

Thanks! I took your suggestions into account and merged.
